From 1e1e64cbbb37eb74e2eef29e2605f0f66350e454 Mon Sep 17 00:00:00 2001
From: Sergio Durigan Junior <sergiodj@chainguard.dev>
Date: Fri, 29 Aug 2025 20:13:40 -0400
Subject: [PATCH 10/11] Always disable Microsoft telemetry

Starting with go 1.25, Microsoft implemented their own telemetry.
Let's disable it entirely.

Signed-off-by: Sergio Durigan Junior <sergiodj@chainguard.dev>
---
 src/cmd/internal/telemetry/counter/counter.go | 33 +++----------------
 1 file changed, 4 insertions(+), 29 deletions(-)

diff --git a/src/cmd/internal/telemetry/counter/counter.go b/src/cmd/internal/telemetry/counter/counter.go
index 2c1b8143d6..416934d0dd 100644
--- a/src/cmd/internal/telemetry/counter/counter.go
+++ b/src/cmd/internal/telemetry/counter/counter.go
@@ -7,18 +7,11 @@
 package counter
 
 import (
-	"cmp"
-	"context"
 	"flag"
-	"log"
-	"log/slog"
 	"os"
-	"time"
 
 	"golang.org/x/telemetry/counter"
 
-	mstelemetry "github.com/microsoft/go-infra/telemetry"
-	msconfig "github.com/microsoft/go-infra/telemetry/config"
 	mscounter "github.com/microsoft/go-infra/telemetry/counter"
 )
 
@@ -39,31 +32,13 @@ func Open() {
 }
 
 func OpenMicrosoft() {
-	if os.Getenv("MS_GOTOOLCHAIN_TELEMETRY_ENABLED") == "0" {
-		// Telemetry disabled.
-		return
-	}
-	var logger *slog.Logger
-	if os.Getenv("MS_GOTOOLCHAIN_TELEMETRY_VERBOSE") == "1" {
-		logger = slog.Default()
-	}
-	mstelemetry.Start(mstelemetry.Config{
-		InstrumentationKey: appInsightsInstrumentationKey,
-		Endpoint:           cmp.Or(os.Getenv("MS_GOTOOLCHAIN_TELEMETRY_ENDPOINT"), appInsightsEndpoint), // configurable for testing purposes
-		AllowGoDevel:       os.Getenv("MS_GOTOOLCHAIN_TELEMETRY_ALLOW_GO_DEVEL") == "1",
-		UploadConfig:       msconfig.Config,
-		Logger:             logger,
-	})
+	// Chainguard: Always disable MS telemetry.
+	return
 }
 
 func CloseMicrosoft() {
-	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
-	defer cancel()
-	start := time.Now()
-	mstelemetry.Close(ctx)
-	if os.Getenv("MS_GOTOOLCHAIN_TELEMETRY_VERBOSE") == "1" {
-		log.Printf("telemetry: closed in %s\n", time.Since(start))
-	}
+	// Chainguard: Always disable MS telemetry.
+	return
 }
 
 // Inc increments the counter with the given name.
-- 
2.51.0

